File: parse-test-xtext-parent/parse-test-xtext/src/test/java/dev/algo/parsetest/xtext/tests/ArithmeticExprParseTest.java
package dev.algo.parsetest.xtext.tests;

import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertTrue;

import org.eclipse.emf.common.util.EList;
import org.eclipse.emf.ecore.resource.Resource.Diagnostic;
import org.eclipse.xtext.testing.InjectWith;
import org.eclipse.xtext.testing.extensions.InjectionExtension;
import org.eclipse.xtext.testing.util.ParseHelper;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;

import com.google.inject.Inject;

import dev.algo.parsetest.xtext.arithmeticExpr.Model;

@ExtendWith(InjectionExtension.class)
@InjectWith(ArithmeticExprInjectorProvider.class)
class ArithmeticExprParseTest {

	@Inject
	ParseHelper<Model> parseHelper;
	
	@Test
	void loadModel() throws Exception {
		Model result = parseHelper.parse("(13 * 7) * (2 * 17)");
		assertNotNull(result, "Unexpected null parse result");
		
		EList<Diagnostic> errors = result.eResource().getErrors();
		assertTrue(errors.isEmpty(), "Unexpected errors: " + errors);
	}

}

File: parse-test-benchmark/src/test/kotlin/dev/algo/parsetest/benchmark/ArithmeticExprParseTest.kt
package dev.algo.parsetest.benchmark

import com.google.inject.Inject
import com.google.inject.Injector
import dev.algo.parsetest.xtext.ArithmeticExprStandaloneSetup
import dev.algo.parsetest.xtext.arithmeticExpr.Model
import org.eclipse.xtext.testing.extensions.InjectionExtension
import org.eclipse.xtext.testing.util.ParseHelper
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith




@ExtendWith(InjectionExtension::class)
internal class ArithmeticExprParseTest {

    private lateinit var parseHelper: ParseHelper<Model>

    @Test
    @Throws(Exception::class)
    fun loadModel() {
        val setup = ArithmeticExprStandaloneSetup()
        val injector: Injector = setup.createInjectorAndDoEMFRegistration()
        val parseHelper = injector.getInstance(ParseHelper::class.java)
        val result = parseHelper.parse("(13 * 7) * (2 * 17)")
        Assertions.assertNotNull(result, "Unexpected null parse result")

        val errors = result.eResource().errors
        Assertions.assertTrue(errors.isEmpty(), "Unexpected errors: $errors")
    }
}
